contracts/WildcardsQV.sol

```diff
@@ -114,32 +142,61 @@ contract WildcardsQV is Initializable {
     ///////////////////////////////////
     /////// vote function  ///////////
     ///////////////////////////////////
-    function vote(uint256 proposalIdToVoteFor, uint256 amount, uint256 sqrt) external {
+    function vote(uint256 proposalIdToVoteFor, uint256 amount, uint256 sqrt)
+        external
+    {
         // Check they are a wildcards user
-        require(wildCardToken.balanceOf(msg.sender)>0, "Does not own a WildCard");
+        require(
+            wildCardToken.balanceOf(msg.sender) > 0,
+            "Does not own a WildCard"
+        );
         // Check they have at least 1 wildcards loyalty token:
         require(amount >= 10**18, " Minimum vote one token");
-        // Check they are voting for a valid proposal: 
-        require(state[proposalIdToVoteFor] == ProposalState.Active, "Proposal not Active");
+        // Check they are voting for a valid proposal:
+        require(
+            state[proposalIdToVoteFor] == ProposalState.Active,
+            "Proposal not Active"
+        );
         // Check that they haven't yet voted for the proposal in this iteration
-        require(!hasUserVotedForProposalIteration[proposalIteration][msg.sender][proposalIdToVoteFor], "Already voted on this proposal");
+        require(
+            !hasUserVotedForProposalIteration[proposalIteration][msg
+                .sender][proposalIdToVoteFor],
+            "Already voted on this proposal"
+        );
         // Send their wildcards tokens to the burn address
-        require(loyaltyToken.transferFrom(msg.sender,burnAddress,amount), "Loyalty Token transfer failed");
+        require(
+            loyaltyToken.transferFrom(msg.sender, burnAddress, amount),
+            "Loyalty Token transfer failed"
+        );
         // Validate the square root
         require(sqrt.mul(sqrt) == amount, "Square root incorrect");
+
+        hasUserVotedForProposalIteration[proposalIteration][msg
```

Comment:
Doing this change before any of the accounting to prevent reentrancy nonsense. Zuck always tells me its best to do these things first :)